src/Microsoft.VisualStudio.Services.Agent/Util/IOUtil.cs

```diff
@@ -333,6 +333,41 @@ public static void CopyDirectory(string source, string target, CancellationToken
             }
         }
 
+        public static void ValidateExecutePermission(string directory)
```

Comment:
Oops my above reply was intended for a different comment.

To address this comment: No, technically it's the "Execute" permission (permission to list contents).